### ISSUE TITLE
sqlite and /seafhttp/repo/head-commits-multi/

### ISSUE THREAD

**Issue by niol** [haiwen/seafile-server#112]
```LOCK IN SHARE MODE``` does not seem valid sqlite.

```
../common/seaf-db.c(425): Error prepare statement SELECT repo_id, commit_id FROM Branch WHERE name='master' AND repo_id IN ('b21cae35-bc19-40e1-b411-abd2ffabbd9b','6e97f20d-8b90-42d3-823b-9c481014e3ef') LOCK IN SHARE MODE: sqlite3_prepare_v2 failed: near "LOCK": syntax error.
```

**Comment by niol** [comment]
This simple patch works around the problem:
```patch
--- seafile-server.orig/server/http-server.c
+++ seafile-server/server/http-server.c
@@ -1112,7 +1112,7 @@ head_commits_multi_cb (evhtp_request_t *
             g_string_append_printf (id_list_str, ",'%s'", json_string_value(id));
     }

-    sql = g_strdup_printf ("SELECT repo_id, commit_id FROM Branch WHERE name='master' AND repo_id IN (%s) LOCK IN SHARE MODE",
+    sql = g_strdup_printf ("SELECT repo_id, commit_id FROM Branch WHERE name='master' AND repo_id IN (%s)",
                            id_list_str->str);
     commit_id_map = json_object();
     if (seaf_db_statement_foreach_row (seaf->db, sql,
```

**Comment by niol** [comment]
This seems to be fixed by bb89603e5b099591ad1c5939d56fdc2306b1829c .

**Comment by shoeper** [comment]
Yes, the issue can be closed.
